industrial_taxonomy/pipeline/glass_house/utils.py

```diff
@@ -0,0 +1,20 @@
+from pandas import DataFrame
+
+
+def get_top_matches(df: DataFrame) -> DataFrame:
+    """For each y get the x and sim_mean corresponding to highest sim_mean."""
+    return df.loc[df.groupby("y").sim_mean.idxmax()]
+
+
+def _filter_cols(df: DataFrame) -> DataFrame:
+    # Optimisation - keep used cols only
+    return df.loc[:, ("x", "y", "sim_mean")]
+
+
+def _drop_poor_matches(df: DataFrame) -> DataFrame:
```

Comment:
I'm not sure what this is doing differently to `get_top_matches` asides from returning a pipe object. If that's needed, could this not be a wrapper of the other func?

Reply:
I think this is an accident of history :sweat_smile: